aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPluginTest.kt

```diff
@@ -472,4 +476,111 @@ class RealAWSCognitoAuthPluginTest {
 
         assertEquals(expectedException, resultCaptor.captured.cause)
     }
+
+    @Test
+    fun `update user attribute fails when not in SignedIn state`() {
+        // GIVEN
+        val onSuccess = mockk<Consumer<AuthUpdateAttributeResult>>(relaxed = true)
+        val onError = mockk<Consumer<AuthException>>(relaxed = true)
+        val listenLatch = CountDownLatch(1)
+
+        currentState = AuthenticationState.NotConfigured()
+
+        every {
+            onError.accept(AuthException.InvalidStateException())
+        } answers {
+            listenLatch.countDown()
+        }
+
+        // WHEN
+        plugin.updateUserAttribute(
+            AuthUserAttribute(AuthUserAttributeKey.email(), "test@test.com"),
+            onSuccess,
+            onError
+        )
+
+        assertTrue { listenLatch.await(5, TimeUnit.SECONDS) }
+        coVerify(exactly = 1) { onError.accept(AuthException.InvalidStateException()) }
+        verify(exactly = 0) { onSuccess.accept(any()) }
+    }
+
+    @Test
+    fun `update user attribute with success`() {
+        // GIVEN
+        val onSuccess = mockk<Consumer<AuthUpdateAttributeResult>>(relaxed = true)
+        val onError = mockk<Consumer<AuthException>>(relaxed = true)
+        val listenLatch = CountDownLatch(1)
+
+        val currentAuthState = mockk<AuthState> {
+            every { authNState } returns AuthenticationState.SignedIn(mockk())
+            every { authZState } returns AuthorizationState.SessionEstablished(credentials)
+        }
+        every { authStateMachine.getCurrentState(captureLambda()) } answers {
+            lambda<(AuthState) -> Unit>().invoke(currentAuthState)
+        }
+
+        coEvery {
+            authService.cognitoIdentityProviderClient?.updateUserAttributes(any<UpdateUserAttributesRequest>())
+        } returns UpdateUserAttributesResponse.invoke {
+        }
+
+        every {
+            onSuccess.accept(any<AuthUpdateAttributeResult>())
+        } answers {
+            listenLatch.countDown()
+        }
+
+        // WHEN
+        plugin.updateUserAttribute(
+            AuthUserAttribute(AuthUserAttributeKey.email(), "test@test.com"),
+            onSuccess,
+            onError
+        )
+
+        assertTrue { listenLatch.await(5, TimeUnit.SECONDS) }
```

Comment:
Also test wether Amplify API response matches the expected response.

Reply:
done